source/Octopus.Server.Extensibility.Authentication.Guest/GuestAuth/GuestCredentialValidator.cs

```diff
@@ -39,7 +42,7 @@ public IUser ValidateCredentials(string username, string password)
             }
             else
             {
-                return user;
+                return new AuthenticationUserCreateOrUpdateResult(user);
             }
 
             return null;
```

Comment:
This `return null` looks out of place since the other return nulls got changed to return an empty object, this intended?